### ISSUE TITLE
max_htlc_msat_ratio to apply on OutBound capacity

### ISSUE THREAD

**Issue by BhaagBoseDK** [accumulator/charge-lnd#48]
max_htlc_msat_ratio (or a new parameter) which can apply on the outbound capacity of the channel instead of total capacity.

**Comment by accumulator** [comment]
This would easily lead to spamming the gossip. 
What use-case are you trying to accomplish with this?

**Comment by BhaagBoseDK** [comment]
max_htlc <= outgoing channel capacity

**Comment by steff123** [comment]
Would it lead to more gossip than a proportional fee strategy?

**Comment by BhaagBoseDK** [comment]
It appears adjusting Max_HTLC is catching up and plebs are applying workarounds - creating custom config files and running charge lnd twice. 
That is doing much harm to gossip and could be avoided by this change request.

https://plebnet.wiki/wiki/Using_Charge-Lnd_to_Adjust_HTLC_Limits

max_htlc_msat_ratio = chan.ratio 
or 
max_htlc_msat = chan.local_balance.

**Comment by accumulator** [comment]
> It appears adjusting Max_HTLC is catching up and plebs are applying workarounds - creating custom config files and running charge lnd twice.
> That is doing much harm to gossip and could be avoided by this change request.

No both are actually unwise. There's no sane reason to set the max HTLC to your current local balance on each run. This is equivalent to publishing your channel ratio, which was intentionally not included in the lightning spec.

It also does not help pathfinding, as this value is most of the time outdated.

sigh, maybe it's better to remove this feature as ppl seem to love shooting themselves in the foot.

**Comment by yuyaogawa** [comment]
Wouldn't it mitigate htlc failure? I understand setting max HTLC to local balance exposes the channel balance, but you cannot make sure if it is actually the channel balance as long as you do not probe the channel (which does not give you 100% accuracy).

**Comment by accumulator** [comment]
Yes, it would mitigate routing failure. But sending out an update on the gossip network after each balance change is not a scalable solution. 

It would not really be a big problem if running charge-lnd once a day, but some people run it every 5 minutes.
Also, if you *are* careful with not running charge-lnd too often, the max HTLC limit could also limit the use of your channel if it refills, as it's out of date.

Still, it IS currently possible to limit your max-htlc, but you'll have to split it up in a few separate rules:
1: local_balance < 100_000 -> max-htlc = 50_000
2: local_balance < 1_000_000 -> max-htlc = 500_000
etc

So in summary, I understand the need to signal to the network that a channel is depleted in order to mitigate routing failures, but it should be done in a way that doesn't flood the network.

**Comment by jasonpearce** [comment]
Good discussion. 

Could a compromise be to also add something like min_max_htlc_delta = 5 to reduce flooding the network with changes? That way, new max_htlc changes would only occur if the delta is 5 percent or greater. 

Something like this:

**max_htlc_msat**
Maximum size (in msat) of HTLC to allow. Use -1 to set to local balance -1% less than local capacity. Use -5 to set to local balance -5% less than local capacity. Etc..

**max_htlc_msat_delta**
Minimum percent change in the max_htlc_msat value before updating a channel. Use 5 to wait for a 5% delta before broadcasting the new max_htlc_msat value to the gossip network. Recommendation is to match the max_htlc_msat value.

Thanks for the consideration.

**Comment by hsjoberg** [comment]
> It also does not help pathfinding, as this value is most of the time outdated.

@accumulator  Please reconsider your stance regarding this feature request.

This article by Rene Pickhardt goes over why nodes in the network should adjust their `htlc_maximum_msat values`, as it leads to a more robust and reliable network:
https://blog.bitmex.com/the-power-of-htlc_maximum_msat-as-a-control-valve-for-better-flow-control-improved-reliability-and-lower-expected-payment-failure-rates-on-the-lightning-network/

@jasonpearce's suggestion would help prevent spam and is a trivial fix.
Really, unless I'm missing something, setting `strategy` to `proportional` appears to be just as concerning and would also benefit from a threshold (if there isn't already one).
